code/modules/atmospherics/machinery/air_alarm/_air_alarm.dm

```diff
@@ -574,28 +574,28 @@ GLOBAL_LIST_EMPTY_TYPED(air_alarms, /obj/machinery/airalarm)
 
 	if(danger_level)
 		alarm_manager.send_alarm(ALARM_ATMOS)
-		if(pressure <= WARNING_LOW_PRESSURE && temp <= BODYTEMP_COLD_WARNING_1+10)
+		if(pressure <= tlv_collection["pressure"]["hazard_min"] && temp <= tlv_collection["temperature"]["hazard_min"])
```

Comment:
You're indexing these incorrectly

tlv_collection is an assoc list `key = /datum/tlv_connection`, meaning you're doing var access shenanigans which is ugly. Should be `tlv_collection["pressure"].hazard_min`, `tlv_collection["temperature"].hazard_min`, etc. 

BUT ALSO quick glance says there's a proc on those datums you should use

So it should be `if(tlv_collection["pressure"].check_value(pressure) && tlv_collection["temperature"].check_value(temperature))`

Reply:
Oh nvm about the proc I guess it doesn't apply here since we're looking for specifics on how dangerous it is, not that it's dangerous in general

But the first part still applies

Reply:
Applied via https://github.com/tgstation/tgstation/pull/86917/commits/f2e9071cfbc6f8cc3b4c6c7227e7157440e32781

I haven't touched DM code in a while. Initially, I was trying to use `tlv_collection.pressure.hazard_min` but the lookup was throwing errors. I can't remember if it's due to it being a datum.
 
I also was aware of `check_value` but saw that it checks for both the min/max at the same time.